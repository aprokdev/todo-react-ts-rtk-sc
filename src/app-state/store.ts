import hideCompletedReducer from '@app-state/hideCompleted/hideCompletedSlice';
import isSavedTodosReducer from '@app-state/isSavedTodos/isSavedTodosSlice';
import sortingTitleReducer from '@app-state/sortingTitle/sortingTitleSlice';
import todosReducer from '@app-state/todos/todosSlice';
import { PreloadedState, combineReducers, configureStore } from '@reduxjs/toolkit';

export const rootReducer = combineReducers({
    todos: todosReducer,
    sortingTitle: sortingTitleReducer,
    hideCompleted: hideCompletedReducer,
    isSavedTodos: isSavedTodosReducer,
});

export function setupStore(preloadedState?: PreloadedState<RootState>) {
    const store = configureStore({
        reducer: rootReducer,
        devTools: true,
        preloadedState,
    });

    let isSavedTodos = true;
    // adds subscribtion on every action:
    store.subscribe(() => {
        // saves current todos and sorting states to LocalStorage only if action type is not "clearLocalStorage":
        isSavedTodos = store.getState().isSavedTodos;
        if (isSavedTodos) {
            localStorage.setItem('listTodos', JSON.stringify(store.getState().todos));
            localStorage.setItem('sortingTitle', JSON.stringify(store.getState().sortingTitle));
        }
        isSavedTodos = true;
    });

    return store;
}

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = ReturnType<typeof setupStore>;
export type AppDispatch = AppStore['dispatch'];
