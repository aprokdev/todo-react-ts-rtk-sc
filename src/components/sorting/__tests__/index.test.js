import React from 'react';
import { Provider } from 'react-redux';
import { act, create } from 'react-test-renderer';
import { setupStore } from '~app-state/store';
import Sorting from '../index';

const store = setupStore();

function Test() {
    return (
        <Provider store={store}>
            <Sorting />
        </Provider>
    );
}

describe('Sorting', () => {
    test('matches snapshot', () => {
        let tree;
        act(() => {
            tree = create(<Test />);
        });
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
