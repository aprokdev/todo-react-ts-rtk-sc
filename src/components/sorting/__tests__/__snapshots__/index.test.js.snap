// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting matches snapshot 1`] = `
<div
  className="sorting"
>
  <h3
    className="sorting__title"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    âœ¨ Sort tasks by: 
    CREATION DATE
  </h3>
</div>
`;
